azurerm/internal/services/appconfiguration/app_configuration_resource.go

```diff
@@ -408,3 +446,31 @@ func flattenAppConfigurationAccessKey(input appconfiguration.APIKey) []interface
 		},
 	}
 }
+
+func expandAppConfigurationIdentity(identities []interface{}) *appconfiguration.ResourceIdentity {
+	if len(identities) == 0 {
+		return &appconfiguration.ResourceIdentity{
+			Type: appconfiguration.None,
+		}
+	}
+	identity := identities[0].(map[string]interface{})
+	identityType := appconfiguration.IdentityType(identity["type"].(string))
+	return &appconfiguration.ResourceIdentity{
+		Type: identityType,
+	}
+}
+func flattenAppConfigurationIdentity(identity *appconfiguration.ResourceIdentity) []interface{} {
+	if identity == nil || identity.Type == appconfiguration.None {
+		return []interface{}{}
+	}
+	result := make(map[string]interface{})
+	result["type"] = identity.Type
+	if identity.PrincipalID != nil {
+		result["principal_id"] = *identity.PrincipalID
+	}
+	if identity.TenantID != nil {
+		result["tenant_id"] = *identity.TenantID
+	}
```

Comment:
can we make this:

```suggestion
	principalId := ""
	if identity.PrincipalID != nil {
		principalId = *identity.PrincipalID
	}
	
	tenantId := ""
	if identity.TenantID != nil {
		tenantId = *identity.TenantID
	}
	
	return []interface{}{
		map[string]interface{}{
			"type": string(identity.Type),
			"principal_id": principalId,
			"tenant_id": tenantId,
		},
	}
```

Which avoids a perpetual diff should the API return a bad response here